java-checks/src/main/java/org/sonar/java/checks/MismatchPackageDirectoryCheck.java

```diff
@@ -27,12 +29,12 @@
 import org.sonar.plugins.java.api.tree.CompilationUnitTree;
 import org.sonar.plugins.java.api.tree.PackageDeclarationTree;
 
-import java.io.File;
-
 @Rule(key = "S1598")
 public class MismatchPackageDirectoryCheck extends BaseTreeVisitor implements JavaFileScanner {
 
   private JavaFileScannerContext context;
+  private static final String MESSAGE = "This file \"${0}\" should be located in \"${1}\" directory, not in \"${2}\"";
```

Comment:
I know that you dream to be a millionaire, but it's not a reason to add *$$$* everywhere.
MessageFormat.format arguments should be `{0}`, `{1}`, `{2}`  and not  `${0}`, `${1}`, `${2}`

Reply:
:sweat_smile: thanks, I mixed two syntaxes